hooks/airtable-conversion-utils.ts

```diff
@@ -18,7 +20,101 @@ function getThumbnailUrl (imageAttachment: any): string|null {
   return largeThumbnail ? largeThumbnail.url : null
 }
 
+class AirtableRecords {
+  protected apiKey: string;
+  private baseId: string;
+  private tableId: string;
+  private view: string;
+
+  constructor (apiKey: string, baseId: string, tableId: string, view: string) {
+    this.apiKey = apiKey
+    this.baseId = baseId
+    this.tableId = tableId
+    this.view = view
+  }
+
+  /**
+   * Get the Airtable field name for a given field ID.
+   *
+   * @param fieldId Field ID
+   * @returns {Promise<string | null>} Field name
+   */
+  public getFieldName (fieldId: string): Promise<string | null> {
```

Comment:
The `getFieldName` method returns a Promise because it is performing an asynchronous operation.

The method uses the Airtable library to make a request to the Airtable API to retrieve the field name for a given field ID. The `eachPage` method used in the method returns a promise that resolves once all the pages of records have been retrieved, so the method returns this promise to the caller.

The `getAllFieldNames` method also returns a promise because it is using Promise.all to wait for all the field name promises returned by the `getFieldName` method to resolve before returning the object containing field names mapped to keys.

`getAllFieldNames` needs to _await_ for the resolution of each `getFieldName` to get the strings, hence they return promises of values since they can not return values at the time they are called.